blue-xdma/src/include/XDMADescriptorGenerator.bsv

```diff
@@ -0,0 +1,159 @@
+import FIFOF::*;
+import Clocks :: *;
+import GetPut :: *;
+import BlueAXI :: *;
+import BlueLib :: *;
+import AXI4LiteMaster :: *;
+import StmtFSM :: *;
+`include "Config.defines"
+
+typedef struct {
+    Bit#(28) length;
+    Bit#(64) srcAddr;
+    Bit#(64) dstAddr;
+} XDMADescriptor deriving (Bits, Eq, FShow);
+
+typedef struct {
+    Bool isC2H;
+    Bool isEnable;
+} ControlDMAIfc deriving (Bits, Eq, FShow);
+
+(* always_ready, always_enabled *)
+interface IfcXDMADescriptorGeneratorFab;
+    method Bool load;
+    method Bit#(64) src_addr;
```

Comment:
use camel case, instead of snake case

Reply:
These variables will be used for direct connection to XDMA, so the variable names must comply with the definition of external names. Therefore, they cannot be modified here.